src/containers/Cfe/index.tsx

```diff
@@ -215,13 +235,16 @@ function mapStateToProps(state: State) {
         thumbnailRoot: selectionStateBranch.selectors.getThumbnailRoot(state),
         showSmallScreenWarning: metadataStateBranch.selectors.getShowSmallScreenWarning(state),
         viewerHeader: getViewerHeader(state),
+        showAlignControl: selectionStateBranch.selectors.getSelected3DCellHasTransform(state),
+        alignActive: selectionStateBranch.selectors.getAlignActive(state),
     };
 }
 
 const dispatchToPropsMap = {
     toggleGallery: selectionStateBranch.actions.toggleGallery,
     setShowSmallScreenWarning: metadataStateBranch.actions.setShowSmallScreenWarning,
     requestFeatureData: metadataStateBranch.actions.requestFeatureData,
+    setAlignActive: selectionStateBranch.actions.setAlignActive,
```

Comment:
this seems like a very high level of the app to be adding this functionality.  I know it works just fine like this but I wonder if there is a way that would keep its implementation closer to the component where it's associated.

Reply:
The docs advise that components may be responsible for their own state if the value:

> 1. is absolutely of no interest to the remainder of the application and never will be;
> 2. is not necessary for "hydrating" the application on initial load;
> 3. and cannot be derived from state that exists in the Redux store.

I buy that this case meets all these conditions, the only one I find remotely uncertain is 1: would there ever be a case where another component cares whether alignment is active?

Reply:
Looking closer at the code, I remember why I did it this way: props passed in to the viewer are computed by a Redux selector (`getPropsForVolumeViewer`) which is unaware of state stored in `Cfe`. It's possible to break this convention and pass extra props in from the component (I already broke the ice with this in an earlier PR by passing the viewer a handler for when the control panel closes). But given that this bit of state is used to derive a prop (`transform`, either zeroed out or populated from `FileInfo` depending on the value of `alignActive`) rather than being immediately passed as a prop itself, it felt more idiomatic to do that derivation in the selector.